tests/components/bayesian/test_binary_sensor.py

```diff
@@ -183,36 +186,49 @@ async def test_sensor_state(hass):
     await hass.async_block_till_done()
 
     hass.states.async_set("sensor.test_monitored", "on")
-
+    await hass.async_block_till_done()
     state = hass.states.get("binary_sensor.test_binary")
 
-    assert [] == state.attributes.get("observations")
-    assert state.attributes.get("probability") == 0.2
-
+    assert state.attributes.get("occurred_observation_entities") == [
+        "sensor.test_monitored"
+    ]
+    assert state.attributes.get("observations")[0]["prob_given_true"] == 0.8
+    assert state.attributes.get("observations")[0]["prob_given_false"] == 0.4
+    assert abs(0.0769 - state.attributes.get("probability")) < 0.01
     assert state.state == "off"
 
     hass.states.async_set("sensor.test_monitored", "off")
     await hass.async_block_till_done()
-    hass.states.async_set("sensor.test_monitored", "on")
```

Comment:
why remove these state changes ?

Reply:
because it triggers events in the bayesian sensor that should be tested ?

Reply:
Does it? The state changes back and forth with no assert statements between?

Reply:
Yes... this function is called https://github.com/home-assistant/core/blob/ab4a9b67809f6c0d9f88c7fea97949aaf5aeee58/homeassistant/components/bayesian/binary_sensor.py#L183

Reply:
There are many other places in the tests that that function is tested in, with appropriate asserts that actually confirm it functions as intended, rather then just confirming it does not throw an unhandled exception.

Very happy to re add those lines of you think it adds anything. To me it just appears to be defunct code - running state changes with no tests that it performs as expected.